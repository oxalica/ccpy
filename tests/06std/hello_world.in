def main():
  print('Hello', 'world')
  print('1 + 1 =', 1 + 1)
  print()

  try:
    a = Animal('Unknow')
    b = Cat('Kitty')
    c = Dog('Mike')
    b.hug()
    c.hug()
    a.hug()
    print('Unreachable')
  except Exception as e:
    print('ERR', repr(e))

class Animal(object):
  def __init__(self, name):
    self.name = name

  def hug(self):
    raise NotImplementedError('Cannot hug an abstract Animal')

class Cat(Animal):
  def __init__(self, name):
    Animal.__init__(self, 'Cat ' + name)

  def hug(self):
    print(self.name, ': Meow')

class Dog(Animal):
  def __init__(self, name):
    Animal.__init__(self, 'Dog ' + name)

  def hug(self):
    print(self.name, ': Wow')

main()
