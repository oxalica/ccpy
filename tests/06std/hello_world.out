Std done

Hello world
1 + 1 = 2

Cat Kitty : Meow
Dog Mike : Wow
ERR NotImplementedError("Cannot hug an abstract Animal",)
Done
